admin/includes/functions/general.php

```diff
@@ -2722,10 +2722,14 @@ function zen_get_shipping_enabled($shipping_module) {
  */
   function zen_get_handler_from_type($product_type) {
     global $db;
-
+    global $messageStack;
+    
     $sql = "select type_handler from " . TABLE_PRODUCT_TYPES . " where type_id = '" . (int)$product_type . "'";
     $handler = $db->Execute($sql);
-    if ($handler->EOF) return 'ERROR: Invalid type_handler. Your product_type settings are wrong, incomplete, or damaged.';
+    if ($handler->EOF) { 
+          $messageStack->add('ERROR: Invalid type_handler. Your product_type settings are wrong, incomplete, or damaged.', 'error');
```

Comment:
Okay, if we're going to display the message to the user, this is a good time to change the message to something more specific, and shorter, like:

`ERROR: Invalid product_type specified.`